### PR TITLE
Add modify window event and stop event to STO

### PR DESCRIPTION
**Pull request by CJP10**
### new features

- Adds a new event `FundraiserWindowModifed`
- Adds a new fundraiser status `ClosedEarly`
### modified logic

- Emit `FundraiserClosed` on stop

### CODE REVIEW

File: pallets/sto/src/lib.rs [hunk 1]
```diff
@@ -174,13 +174,23 @@ decl_event!(
...
         /// (primary issuance agent, fundraiser id)
         FundraiserFrozen(IdentityId, u64),
-        /// An fundraiser has been unfrozen.
+        /// A fundraiser has been unfrozen.
         /// (primary issuance agent, fundraiser id)
         FundraiserUnfrozen(IdentityId, u64),
-        /// An fundraiser has been stopped.
+        /// A fundraiser window has been modified.
+        /// (primary issuance agent, fundraiser id, old_start, old_end, new_start, new_end)
+        FundraiserModifyWindow(
```
Comment (maxsam4) [inline pallets/sto/src/lib.rs:185]: For consistency:
```suggestion
        FundraiserWindowModified(
```

File: pallets/sto/src/lib.rs [hunk 2]
```diff
@@ -498,6 +508,7 @@ decl_module! {
                 if let Some(end) = end {
                     ensure!(start < end, Error::<T>::InvalidOfferingWindow);
                 }
+                Self::deposit_event(RawEvent::FundraiserModifyWindow(did, fundraiser_id, fundraiser.start, fundraiser.end, start, end));
```
Comment (maxsam4) [inline pallets/sto/src/lib.rs:511]:
```suggestion
                Self::deposit_event(RawEvent::FundraiserWindowModified(did, fundraiser_id, fundraiser.start, fundraiser.end, start, end));
```

File: pallets/sto/src/lib.rs [hunk 3]
```diff
@@ -487,7 +497,7 @@ decl_module! {
         /// `1_000` placeholder
         #[weight = <T as Trait>::WeightInfo::modify_fundraiser_window()]
         pub fn modify_fundraiser_window(origin, offering_asset: Ticker, fundraiser_id: u64, start: T::Moment, end: Option<T::Moment>) -> DispatchResult {
-            Self::ensure_perms_pia(origin, &offering_asset)?;
+            let did = Self::ensure_perms_pia(origin, &offering_asset)?.0;
```
Comment (Centril) [inline pallets/sto/src/lib.rs:500]:
```suggestion
            let did = Self::ensure_perms_pia(origin, &offering_asset)?.0.for_event();
```

**Review by satyamakgec — APPROVED** [review]
Approved! build is failing though.

**Review by fmiguelgarcia — APPROVED** [review]

**Review by maxsam4 — COMMENTED** [review]
The schema needs to be updated as well

**Review by maxsam4 — APPROVED** [review]

**Pull request merged**